api/redhatopenshift/HcpCluster.Management/hcpCluster-models.tsp

```diff
@@ -486,3 +528,101 @@ model HcpOpenShiftClusterAdminCredential {
  *   End HCP cluster credentials
  * =======================================
  */
+
+/*
+ * =======================================
+ *  ExternalAuth resources
+ * =======================================
+ */
+
+/** External Auth profile */
+model ExternalAuthProfileProperties {
+  /** Provisioning state */
+  @visibility(Lifecycle.Read)
+  provisioningState?: ExternalAuthProvisioningState;
```

Comment:
I just noticed this is marked optional but yet it's read-only.  And we do the same for cluster and node pool provisioning state.  AFAIU, `?` doesn't make sense on a read-only field.  It's not causing problems it's just pointless.  Let's drop it here and I'll do a sweep of the rest of the API.

Reply:
Does that mean it has to be there in the Read response though? Optionally, returning something seems valid to me.

Reply:
This could be one of those things were I _thought_ I understood TypeSpec but I'm wrong, but I believe the optionality of a field only applies to creation (PUT) requests.  Fields that aren't always returned in a response I don't think have to be reflected in the API model.

`provisioningState` specifically is always present on resources where async operations are defined.

Reply:
I got a typespec error "The provisioningState property must be optional." so leaving optional